### PR TITLE
Indicate when "Prepare" button has already been clicked

### PR DESCRIPTION
**Pull request by ianpaschal**
Contributes to CURA-5551

### CODE REVIEW

File: resources/qml/SaveButton.qml [hunk 1]
```diff
@@ -48,14 +48,15 @@ Item {
...
     function sliceOrStopSlicing() {
-        try {
-            if ([1, 5].indexOf(base.backendState) != -1) {
-                CuraApplication.backend.forceSlice();
-            } else {
-                CuraApplication.backend.stopSlicing();
-            }
-        } catch (e) {
-            console.log('Could not start or stop slicing', e)
+        if ( [ 1, 5 ].indexOf( base.backendState ) != -1 )
```
Comment (diegopradogesto) [inline resources/qml/SaveButton.qml]: Code style: remove spaces

File: resources/qml/SaveButton.qml [hunk 2]
```diff
@@ -48,14 +48,15 @@ Item {
...
+        {
+            prepareButton.preparingToSlice = true;
+            CuraApplication.backend.forceSlice();
+        }
+        else
+        {
+            prepareButton.preparingToSlice = false;
+            CuraApplication.backend.stopSlicing();
```
Comment (diegopradogesto) [inline resources/qml/SaveButton.qml]: Why was the try/catch statement removed? That was probably added for a reason, to catch special situations.

File: resources/qml/SaveButton.qml [hunk 3]
```diff
@@ -180,7 +181,23 @@ Item {
             anchors.rightMargin: UM.Theme.getSize("sidebar_margin").width
 
             // 1 = not started, 4 = error, 5 = disabled
-            text: [1, 4, 5].indexOf(base.backendState) != -1 ? catalog.i18nc("@label:Printjob", "Prepare") : catalog.i18nc("@label:Printjob", "Cancel")
+            text: {
+                if ( preparingToSlice )
```
Comment (diegopradogesto) [inline resources/qml/SaveButton.qml]: Code style: remove spaces

File: resources/qml/SaveButton.qml [hunk 4]
```diff
@@ -180,7 +181,23 @@ Item {
...
+                {
+                    return catalog.i18nc("@label:Printjob", "Preparing");
+                }
+                else
+                {
+                    if ( [ 1, 4, 5 ].indexOf( base.backendState ) != -1 )
```
Comment (diegopradogesto) [inline resources/qml/SaveButton.qml]: Code style: remove spaces

File: resources/qml/SaveButton.qml [hunk 5]
```diff
@@ -237,6 +254,18 @@ Item {
                 }
                 label: Item { }
             }
+
+            Connections {
+                target: UM.Backend
+                onStateChanged:
+                {
+                    if ( [ 2, 3 ].indexOf( UM.Backend.state) != -1 )
```
Comment (diegopradogesto) [inline resources/qml/SaveButton.qml]: Code style: remove spaces

File: resources/qml/SaveButton.qml [hunk 6]
```diff
@@ -237,6 +254,18 @@ Item {
...
+                    {
+                        prepareButton.preparingToSlice = false;
+                    }
+
```
Comment (diegopradogesto) [inline resources/qml/SaveButton.qml]: Remove empty line.

File: resources/qml/SaveButton.qml (hunk 2 above)
Comment (ianpaschal) [reply resources/qml/SaveButton.qml]: Chris added it and we don't do that anywhere else nor were we really handling the errors. This logic should be done on the Python side and use the proper logger class.

**Comment by ianpaschal** [comment]
@diegopradogesto left a comment regarding the try/catch. Everything else is fixed

**Pull request merged**